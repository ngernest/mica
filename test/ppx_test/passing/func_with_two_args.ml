module type S = sig
  type 'a t

  val f : int -> 'a t -> 'a t
end
[@@deriving_inline mica_types]

type expr = F of int * expr
type ty = IntT

let gen_expr ty = let open Core.Quickcheck.Generator.Let_syntax in "TODO"
let _ = gen_expr

[@@@end]
