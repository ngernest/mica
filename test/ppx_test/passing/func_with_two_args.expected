module type S  = sig type 'a t val f : int -> 'a t -> 'a t end[@@deriving
                                                                mica_types]
include
  struct
    type expr =
      | F of int * expr 
    type ty =
      | IntT 
    let gen_expr ty =
      let open Core in
        let open Quickcheck.Generator in
          let open Let_syntax in
            size >>=
              (fun x ->
                 match ty with | IntT -> of_list [F (int__001_, expr__002_)])
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
